src/registry/checkbox/edit-validation.ts

```diff
@@ -0,0 +1,15 @@
+import {IntlShape} from 'react-intl';
+import {z} from 'zod';
+
+import {buildCommonSchema} from '@/registry/validation';
+
+// undefined (optional) for unspecified, otherwise a finite numeric value. Note that
+// null would be nicer, but formio's schema does not support null for validate.min,
+// validate.max or defaultValue
+const booleanSchema = z.boolean().optional();
+
+const defaultValueSchema = z.object({defaultValue: booleanSchema});
```

Comment:
If you set `EditForm.defaultValues = {...}` appropriately, I think it'll be pinned to either `false` or `true` and will add in any missing `defaultValue` key.

I think we can just stick to `buildCommonSchema` for checkbox component  - the UI control is a checkbox anyway which forces boolean value.

Reply:
So it resutls in `const schema = (intl: IntlShape) => buildCommonSchema(intl);`?

Reply:
yep!

At some point we should set up the `useZodErrorMap` hook from the SDK here too, but it's still a bit mixed to set up more specific error messages that need to be translatable. And we need to check how many translations overlap, potentially that should also become its own library.